src/providers/pronouns/alejo/PronounsAlejoApi.cpp

```diff
@@ -0,0 +1,124 @@
+#include "providers/pronouns/alejo/PronounsAlejoApi.hpp"
+
+#include "common/network/NetworkRequest.hpp"
+#include "common/network/NetworkResult.hpp"
+#include "common/QLogging.hpp"
+#include "providers/pronouns/UserPronouns.hpp"
+
+namespace chatterino::pronouns {
+
+UserPronouns AlejoApi::parse(const QJsonObject &object)
+{
+    if (!this->pronounsFromId.has_value())
+    {
+        return {};
+    }
+
+    auto pronoun = object["pronoun_id"];
+
+    if (!pronoun.isString())
+    {
+        return {};
+    }
+
+    auto pronounStr = pronoun.toString();
+    std::shared_lock lock(this->mutex);
+    auto iter = this->pronounsFromId->find(pronounStr);
+    if (iter != this->pronounsFromId->end())
+    {
+        return {iter->second};
+    }
+    return {};
+}
+
+AlejoApi::AlejoApi()
+{
+    std::shared_lock lock(this->mutex);
+    if (this->pronounsFromId)
+    {
+        return;
+    }
+
+    qCDebug(chatterinoPronouns) << "Fetching available pronouns for alejo.io";
+    NetworkRequest(AlejoApi::API_URL + AlejoApi::API_PRONOUNS)
```

Comment:
I would prefer if this network request is only made if the pronoun setting is enabled.
For reference, check out the `BttvEmotes` ctor.
If you want the request to be made in the constructor, which I think is fine for a network request this small, leave the `autoInvoke` param out (or set to true, its default value)